### PR TITLE
Don't report `import-private-name` for relative imports

### LINKED ISSUE #7078: False `import-private-name` positive for relative imports
### Bug description

# pylint: disable=disallowed-name,relative-beyond-top-level,unused-import,missing-module-docstring
from .models import _Private
from ..models import _OtherPrivate

### Configuration

_No response_

### Command used

```shell
pylint --enable-all-extensions foo.py
```

### Pylint output

```shell
************* Module foo
foo.py:2:0: C2701: Imported private object (_Private) (import-private-name)
foo.py:3:0: C2701: Imported private object (_OtherPrivate) (import-private-name)
```

### Expected behavior

No `import-private-name` messages.

I think relative imports should be exempt from this check because they do not tend to refer to external things which are documented and intended to not get this report.

### Pylint version

```shell
pylint 2.14.3
astroid 2.11.6
Python 3.8.12 (default, Sep  4 2021, 14:14:03) 
[GCC 9.3.0]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by scop**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|     | :sparkles: New feature |
|     | :hammer: Refactoring   |
|     | :scroll: Docs          |

## Description

Closes #7078

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
> Wow, ^that "Effect of this PR" report is awesome 

Yeah credit to @DanielNoord :) It's the first time someone fix SO much false positives at once too.

File: doc/whatsnew/2/2.15/index.rst [hunk 1]
```diff
@@ -39,6 +39,10 @@ False positives fixed
 
   Closes #4951
 
+* Don't report ``import-private-name`` for relative imports.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2/2.15/index.rst:42]: I think we can backport in 2.14.4, could you move the changelog please ?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Small change, big result, that's great 👌

File: doc/whatsnew/2/2.15/index.rst (hunk 1 above)
Comment (scop) [reply doc/whatsnew/2/2.15/index.rst]: Sure, done.

**Pull request merged**